jetty-core/jetty-websocket/jetty-websocket-core-client/src/main/java/org/eclipse/jetty/websocket/core/client/CoreClientUpgradeRequest.java

```diff
@@ -247,9 +247,19 @@ public void onComplete(Result result)
         }
 
         URI requestURI = result.getRequest().getURI();
+        Request request = result.getRequest();
         Response response = result.getResponse();
-        int responseStatusCode = response.getStatus();
-        String responseLine = responseStatusCode + " " + response.getReason();
+        int status = response.getStatus();
+        String responseLine = status + " " + response.getReason();
+
+        if ((request.getVersion() == HttpVersion.HTTP_2 && status != HttpStatus.OK_200) ||
+            (request.getVersion() == HttpVersion.HTTP_1_1 && status != HttpStatus.SWITCHING_PROTOCOLS_101))
```

Comment:
Can you replace this with a boolean field that gets set when the upgrade happens, and here just do `if (!upgraded)`?

In this way will work for HTTP/3 as well, etc.